File: CodeArchieve/UOJ/Test-Round-1/83-水题出题人/problem_83/selection_sort.cpp
#include <stdio.h>

int counter, n;
int a[1000001];

void selection_sort_main(){
	int i, j, min, minj, temp;
	counter++;
	scanf("%d", &n);
	counter++;
	for(i = 1; i <= n; i++){
		counter++;
		scanf("%d", &a[i]);
	}
	counter++;
	for(i = 1; i <= n; i++){
		counter++;
		min = 2147483647;
		counter++;
		for(j = i; j <= n; j++){
			counter++;
			if(a[j] < min){
				counter++;
				min = a[j];
				counter++;
				minj = j;
			}
		}
		counter++;
		temp = a[minj];
		counter++;
		a[minj] = a[i];
		counter++;
		a[i] = temp;
	}
	counter++;
	for(i = 1; i <= n; i++){
		counter++;
		//printf("%d ", a[i]);
	}
}

int main(){
	counter = 0;
	selection_sort_main();
	printf("\n");
	if(counter <= 2000000){
		printf("Accepted! The counter value is :%d\n", counter);
	}else{
		printf("Time Limit Exceeded because the counter value is :%d\n", counter);
	}
	return 0;
}

File: CodeArchieve/UOJ/Test-Round-1/83-水题出题人/problem_83/selection_sort.pas
program selection_sort;

var counter, n :longint;
var a :array[1..1000000] of longint;

procedure selection_sort_main;
var i, j, min, minj, temp :longint;
begin
	inc(counter);
	readln(n);
	inc(counter);
	for i := 1 to n do begin
		inc(counter);
		readln(a[i]);
	end;
	inc(counter);
	for i := 1 to n do begin
		inc(counter);
		min := maxlongint;
		inc(counter);
		for j := i to n do begin
			inc(counter);
			if a[j] < min then begin
				inc(counter);
				min := a[j];
				inc(counter);
				minj := j;
			end;
		end;
		inc(counter);
		temp := a[minj];
		inc(counter);
		a[minj] := a[i];
		inc(counter);
		a[i] := temp;
	end;
	inc(counter);
	for i := 1 to n do begin
		inc(counter);
		write(a[i], ' ');
	end;
end;

begin
	counter := 0;
	selection_sort_main;
	writeln;
	if counter <= 2000000 then begin
		writeln('Accepted! The counter value is :', counter);
	end else begin
		writeln('Time Limit Exceeded because the counter value is :', counter);
	end;
end.